From 12102412f27dd8aa655f9c6e446f77f76a53b54c Mon Sep 17 00:00:00 2001
Message-Id: <12102412f27dd8aa655f9c6e446f77f76a53b54c.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Mon, 10 Nov 2014 19:15:58 +0530
Subject: [PATCH 36/75] MUST_REBASE [VPG]: adf/intel/vlv/dsi: Support for
 executing Veriosn 3 sequneces

Add support for parsing Version 3 of MIPI sequence block

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Id8e0cf2e397806c9a564f037dd14e01e36cc78c9
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
---
 .../core/common/dsi/panels/intel_dsi_panel_vbt.c   |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
index 680b4de..10b7dd0 100644
--- a/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
+++ b/drivers/video/adf/intel/core/common/dsi/panels/intel_dsi_panel_vbt.c
@@ -291,14 +291,18 @@ static void generic_exec_sequence(struct dsi_pipe *dsi_pipe, char *sequence)
 	if (!sequence)
 		return;
 
-	pr_debug("Starting MIPI sequence - %s\n", seq_name[*data]);
+	pr_err("Starting MIPI sequence - %s\n", seq_name[*data]);
 
 	/* go to the first element of the sequence */
 	data++;
 
+	if (dsi_pipe->config.dsi->seq_version >= 3)
+		data = data + 4;
+
 	/* parse each byte till we reach end of sequence byte - 0x00 */
 	while (1) {
 		index = *data;
+		pr_debug("ADF: %s: Element Type = %d\n", __func__, index);
 		mipi_elem_exec = exec_elem[index];
 		if (!mipi_elem_exec) {
 			pr_err("Unsupported MIPI element, skipping sequence execution\n");
@@ -308,6 +312,9 @@ static void generic_exec_sequence(struct dsi_pipe *dsi_pipe, char *sequence)
 		/* goto element payload */
 		data++;
 
+		if (dsi_pipe->config.dsi->seq_version >= 3)
+			data++;
+
 		/* execute the element specific rotines */
 		data = mipi_elem_exec(dsi_pipe, data);
 
-- 
1.7.9.5

